### PR TITLE
[Features] alternate theme feature

### PR DESCRIPTION
**Pull request by ahammer**
## Purpose

* Allows selection of default or alternate theme when launching the call Composite

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

1) Select feature from checklist (Secondary theme)
2) Make sure the primary color is different (should be teal).

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: azure-communication-ui/azure-communication-ui/src/main/res/values/azure_communication_ui_feature_flags.xml [hunk 1]
```diff
@@ -11,9 +11,9 @@
     <bool name="azure_communication_ui_feature_flag_bluetooth_audio">false</bool>
 
 
-    <!-- For Tests -->
+    <!-- For Tests & Defaults -->
     <string name="azure_communication_ui_feature_flag_test_label">Test</string>
-    <bool name="azure_communication_ui_feature_flag_test_false">false</bool>
-    <bool name="azure_communication_ui_feature_flag_test_true">true</bool>
+    <bool name="azure_communication_ui_feature_flag_default_false">false</bool>
```
Comment (jamchengms) [inline azure-communication-ui/azure-communication-ui/src/main/res/values/azure_communication_ui_feature_flags.xml:16]: this is a bit odd to me
normally, i would expect
<bool name="azure_communication_ui_feature_flag_default">false</bool>
what is the reason behind using _false and _true

Comment (ahammer) [reply azure-communication-ui/azure-communication-ui/src/main/res/values/azure_communication_ui_feature_flags.xml:16]: Need resource id's for the default values. 

If most features will be false by default, it makes sense to have a default key we can just re-use (less chance of accidental toggling).

Originally needed resource id's for tests, hence the "test" prefix before.

**Comment by ahammer** [comment]
going to amend this PR a bit with updated theme before resubmitting for review.

File: azure-communication-ui/azure-communication-ui/src/main/res/values/azure_communication_ui_feature_flags.xml (hunk 1 above)
Comment (ahammer) [reply azure-communication-ui/azure-communication-ui/src/main/res/values/azure_communication_ui_feature_flags.xml:16]: It's updated in the branch to have 1 flag per feature, the defaults are back to test_ prefix, and only used in text

File: azure-communication-ui/azure-communication-ui/src/main/res/drawable/azure_communication_ui_cancel_leave_call_button.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <solid android:color="@color/azure_communication_ui_color_button_background_transparent" />
     <stroke
         android:width="2dp"
-        android:color="?attr/azure_communication_ui_calling_primary_color"
```
Comment (iaulakh) [inline azure-communication-ui/azure-communication-ui/src/main/res/drawable/azure_communication_ui_cancel_leave_call_button.xml:12]: Adam, this key is public and reviewed by Azure team. what com_ indicates?

Comment (ahammer) [reply azure-communication-ui/azure-communication-ui/src/main/res/drawable/azure_communication_ui_cancel_leave_call_button.xml:12]: naming comes from fluent ui spec, I believe it stands for communication. I'll update documentation in the attrs, theme and custom theme files.

**Review by phileo99 — CHANGES_REQUESTED** [review]
getting this error:
`e: /Users/AlbertLo/projects/microsoft/azure/communication-ui-library-android-publicPreview/azure-communication-ui/azure-communication-ui/src/androidTest/java/com/azure/android/communication/ui/utilities/FeatureFlagsTest.kt: (77, 9): No value passed for parameter 'defaultBooleanId'
`

File: azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/CallLauncherActivity.kt [hunk 3]
```diff
@@ -27,7 +27,9 @@ class CallLauncherActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         // Register Memory Viewer with FeatureFlags
-        initializeMemoryViewFeature(this)
+        conditionallyRegisterDiagnostics(this)
+        FeatureFlags.registerAdditionalFeature(AdditionalFeatures.secondaryThemeFeature)
+
         // Initialize the FeatureFlags enum
```
Comment (iaulakh) [inline azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/CallLauncherActivity.kt:33]: Adam, I think we do nor need code comments as code is very clear? please share your thoughts.

File: azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/features/AdditionalFeatures.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+import android.content.Context
+import android.os.Build
+import com.azure.android.communication.ui.callingcompositedemoapp.R
+import com.azure.android.communication.ui.callingcompositedemoapp.diagnostics.FpsDiagnostics
+import com.azure.android.communication.ui.callingcompositedemoapp.diagnostics.MemoryViewer
+import com.azure.android.communication.ui.utilities.FeatureFlagEntry
+import com.azure.android.communication.ui.utilities.FeatureFlags
+
+// / Extra rules for enabling the Diagnostics feature
```
Comment (iaulakh) [inline azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/features/AdditionalFeatures.kt:11]: Code is very clear :) I think place where we are using comments, we can follow // .

**Review by iaulakh — CHANGES_REQUESTED** [review]

**Review by phileo99 — DISMISSED** [review]
Tested, ran Espresso Tests, looks good
The exception upon returning to home screen will be dealt with in separate PR
👍

File: azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/features/AdditionalFeatures.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package com.azure.android.communication.ui.callingcompositedemoapp.features
+
+import android.content.Context
```
Comment (iaulakh) [inline azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/features/AdditionalFeatures.kt:3]: we may add copyright

**Comment by iaulakh** [comment]
Adam,
For memory and fps diagnostics, on first install, immediately appear on top permissions is open. By default we keep it false, so then when checkbox turned on, permissions are requested.

Noticed warning
Screen Shot 2022-02-23 at 2 46 03 AM

API 21, button tint is not working 
Screen Shot 2022-02-23 at 2 50 10 AM

**Comment by ahammer** [comment]
> Adam, For memory and fps diagnostics, on first install, immediately appear on top permissions is open. By default we keep it false, so then when checkbox turned on, permissions are requested.
> 
> Noticed warning Screen Shot 2022-02-23 at 2 46 03 AM
> 
> API 21, button tint is not working Screen Shot 2022-02-23 at 2 50 10 AM

**Comment by ahammer** [comment]
Closing and updating (as meetings progress, some changes required)

1) Limit to 3 colors used
2) Name the colors more effectively.

Also need to test across wide api range a bit more.

**Comment by ahammer** [comment]
Should be updated/opened again shortly.

**Review by iaulakh — APPROVED** [review]
tested

**Review by ShaunaSong — APPROVED** [review]

**Pull request merged**